pkg/collect/configmap.go

```diff
@@ -49,8 +51,20 @@ func (c *CollectConfigMap) Collect(progressChan chan<- interface{}) (CollectorRe
 	output := NewResult()
 
 	configMaps := []corev1.ConfigMap{}
+	namespace := c.Collector.Namespace
+
 	if c.Collector.Name != "" {
-		configMap, err := c.Client.CoreV1().ConfigMaps(c.Collector.Namespace).Get(c.Context, c.Collector.Name, metav1.GetOptions{})
+		if namespace == "" {
+			kubeconfig := k8sutil.GetKubeconfig()
+			ns, _, err := kubeconfig.Namespace()
+			klog.V(2).Infof("no namespace was set for configmap '%s': using namespace '%s' from current kubeconfig context", c.Collector.Name, ns)
+			if err != nil {
+				return nil, errors.Wrapf(err, "a namespace was not specified for configmap '%s' and could not be discovered from kubeconfig", c.Collector.Name)
+			}
+			namespace = ns
+		}
+		klog.V(1).Infof("looking for configmap '%s' in namespace '%s'", c.Collector.Name, namespace)
```

Comment:
Within the collectors, this needs to be `klog.V(2)` as per [this "very easy to find" documentation](https://github.com/replicatedhq/troubleshoot/blob/a5628874b08c76a8544dfba6da62c7605118d91f/pkg/logger/logger.go#L4)

Note to self: I really need to extract this into our [dev guide documentation](https://github.com/replicatedhq/troubleshoot/issues/1031)